src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs

```diff
@@ -102,6 +102,29 @@ public static TypedData ToRpc(this object value)
                     }
                 }
 
+                // parse ClaimsPrincipal if exists
+                if (request.HttpContext?.User?.Identities != null)
+                {
+                    foreach (var id in request.HttpContext.User.Identities)
+                    {
+                        // asdf
+                        var rpcId = new RpcClaimsIdentity();
+                        if (id.AuthenticationType != null)
+                        {
+                            rpcId.AuthenticationType = id.AuthenticationType;
+                        }
+                        rpcId.NameClaimType = id.NameClaimType;
+                        rpcId.RoleClaimType = id.RoleClaimType;
+
+                        foreach (var claim in id.Claims)
+                        {
+                            rpcId.Claims.Add(new RpcClaim { Value = claim.Value, Type = claim.Type });
```

Comment:
Is validation needed on any of these properties i.e if a  property is null/empty does the identity become invalid? If yes, add validation and also add log if an identity is invalid.

Reply:
Validation here should, I think, be specific to creating a properly-formatted gRPC object; validation of the ClaimsPrincipal's semantics, if any, would be better off centralized in the HTTP extension's logic, which does the bulk of the work and would apply to both in-proc and out-of-proc uses, or in EasyAuth, since this binding is tied to EA being enabled.